src/dialects/postgres/index.js

```diff
@@ -262,6 +262,33 @@ assign(Client_PG.prototype, {
     }
     return resp;
   },
+
+  canCancelQuery: true,
+  cancelQuery(connectionToKill) {
+    const acquiringConn = this.acquireConnection();
+
+    // Error out if we can't acquire connection in time.
+    // Purposely not putting timeout on `pg_terminate_backend` execution because erroring
+    // early there would release the `connectionToKill` back to the pool with
+    // a `KILL QUERY` command yet to finish.
+    return acquiringConn.then((conn) => {
+      return this._wrappedCancelQueryCall(conn, connectionToKill).finally(
+        () => {
+          // NOT returning this promise because we want to release the connection
+          // in a non-blocking fashion
+          this.releaseConnection(conn);
+        }
+      );
+    });
+  },
+  _wrappedCancelQueryCall(conn, connectionToKill) {
+    return this.query(conn, {
+      method: 'raw',
+      sql: 'SELECT pg_terminate_backend(?);',
```

Comment:
Should this be using `pg_cancel_backend` instead of `pg_terminate_backend`?  `pg_terminate_backend` is more heavy handed than necessary most of the time.